framework/src/main/java/org/checkerframework/framework/source/DiagMessage.java

```diff
@@ -36,6 +41,23 @@ public class DiagMessage {
    * @param args the arguments that will be interpolated into the localized message
    */
   public DiagMessage(Diagnostic.Kind kind, @CompilerMessageKey String messageKey, Object... args) {
+    this(null, kind, messageKey, args);
+  }
+
+  /**
+   * Create a DiagMessage.
+   *
+   * @param source where to report the diagnostic
+   * @param kind the kind of message
+   * @param messageKey the message key
+   * @param args the arguments that will be interpolated into the localized message
+   */
+  public DiagMessage(
```

Comment:
This is never called with a non-null `source` argument.  What is you plan for using it?

Reply:
Previously, #5862 used it, but that is no longer the case.  I can remove this feature if you like -- let me know your preference.

Reply:
I'd rather not add code that isn't used.

Reply:
This pull request no longer makes important changes.  It just adds one convenience method.  I'm happy to merge it or discard it -- whichever you prefer.